### ISSUE TITLE
Cannot `mv` servo if built at least once.

### ISSUE THREAD

**Issue by gatoWololo** [servo/servo#23817]
It seems servo hardcodes some build paths somewhere. Once servo has been built once, moving the root file or renaming it, causes servo not to build anymore.

Steps to reproduce:
1) Successfully build Servo once.
2) `mv servo/ servo2`
3) ` ./mach build -d`:
```
error: couldn't read /home/gatowololo/Mozilla/AltServo/servo/components/style/properties/computed_value_flags.rs: No such file or directory (os error 2)
  --> /home/gatowololo/Mozilla/AltServo/servo2/target/debug/build/style-75d6813e48c13779/out/properties.rs:57:9
   |
57 | pub mod computed_value_flags;
   |         ^^^^^^^^^^^^^^^^^^^^

error: aborting due to previous error
```
Notice in the error message above it is still looking under the old path.

**Comment by SimonSapin** [comment]
This looks like a limitation of rustc, possibly related to incremental compilation. Try starting over (`./mach clean`) then compiling with a `CARGO_INCREMENTAL=0` environment variable. Builds after the first will likely be slower.

But really, this doesn’t sound like a scenario worth spending time to support IMO.

**Comment by gatoWololo** [comment]
I agree. I just wanted to document it, as it wasn't obvious at first why servo suddenly stopped building for me.
